api/v1alpha1/konfiguration_types.go

```diff
@@ -27,66 +27,84 @@ const (
 	KonfigureOperatorFinalizer = "finalizers.giantswarm.io/konfigure-operator"
 )
 
-// KonfigurationSpec defines the desired state of Konfiguration.
+// KonfigurationSpec defines the desired state of the Konfiguration.
 type KonfigurationSpec struct {
-	// Targets
+	// Define what konfigurations to render.
 	// +required
 	Targets Targets `json:"targets"`
 
-	// Destination
+	// Defines where and how to store the rendered konfigurations.
 	// +required
 	Destination Destination `json:"destination"`
 
-	// Reconciliation
+	// Defines how to reconcile the Konfiguration.
 	Reconciliation Reconciliation `json:"reconciliation,omitempty"`
 
-	// Sources
+	// Defines where to find the source of the konfiguration that needs to be rendered.
 	Sources Sources `json:"sources"`
 }
 
+// Targets define information on what konfiguration to render and how to render them.
 type Targets struct {
-	// Schema
+	// Defines where to locate the KonfigurationSchema.
 	// +required
 	Schema Schema `json:"schema"`
 
-	Defaults   Defaults             `json:"defaults"`
+	// Define default inputs to be used for every single iteration.
+	// Individual iterations may override default inputs.
+	Defaults Defaults `json:"defaults"`
+
+	// Defines what konfigurations to render. A single reconciliation loop iterates over each entry
+	// and renders the konfiguration, wraps them to Kubernetes manifests and enforces the state of those in the cluster.
 	Iterations map[string]Iteration `json:"iterations,omitempty"`
 }
 
+// Schema defines information on what konfiguration schema to use to render the target konfigurations.
 type Schema struct {
-	// Reference
+	// Defines where to locate the KonfigurationSchema as a Kubernetes manifest.
 	Reference SchemaReference `json:"reference"`
 }
 
+// SchemaReference defines information on how to locate a KonfigurationSchema as a Kubernetes manifest.
 type SchemaReference struct {
-	// Name
+	// Defines the .metadata.name of the KonfigurationSchema Kubernetes manifest to fetch.
 	Name string `json:"name"`
 
-	// Namespace
+	// Defines the .metadata.namespace of the KonfigurationSchema Kubernetes manifest to fetch.
 	Namespace string `json:"namespace"`
 }
 
+// Defaults define information shared across iteration to render konfigurations.
 type Defaults struct {
+	// Defines default variable inputs to be used for every single iteration.
 	Variables []NameValuePair `json:"variables,omitempty"`
 }
 
+// Iteration defines information needed to a single konfiguration to render.
 type Iteration struct {
+	// Defines variable inputs specific for the given iteration.
+	// These variables are merged on top of the default variables, and thus may choose to override default ones.
```

Comment:
"may choose to override" sounds a bit vague. What determines if it overrides or not?